apps/coscrad-frontend/src/utils/generic-components/presenters/id-info-icon.tsx

```diff
@@ -1,16 +1,77 @@
 import { ResourceType } from '@coscrad/api-interfaces';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
 import InfoIcon from '@mui/icons-material/Info';
-import { Box, Tooltip } from '@mui/material';
+import {
+    Dialog,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    IconButton,
+    Tooltip,
+    Typography,
+} from '@mui/material';
+import { useState } from 'react';
 
 interface IdInfoIconProps {
     id: string;
     type: ResourceType;
 }
 
-export const IdInfoIcon = ({ id, type }: IdInfoIconProps): JSX.Element => (
-    <Box component="span">
-        <Tooltip title={`ID: ${type}/${id}`}>
-            <InfoIcon />
-        </Tooltip>
-    </Box>
-);
+export const IdInfoIcon = ({ id, type }: IdInfoIconProps): JSX.Element => {
+    const [openDialog, setOpenDialog] = useState(false);
+
+    const [showTooltip, setShowTooltip] = useState(false);
+
+    const handleDialogOpen = () => {
+        setOpenDialog(true);
+    };
+
+    const handleDialogClose = () => {
+        setOpenDialog(false);
+    };
+
+    const handleCopyClick = () => {
+        navigator.clipboard.writeText(id);
```

Comment:
We could have two buttons, one to copy the id, and one to copy the composite id.

Reply:
If it's easy to do that now, go for it. If it introduces any friction, let's hold off.

Reply:
<img width="504" alt="image" src="https://user-images.githubusercontent.com/63014783/233738063-e7c1007c-f766-4a68-80ec-91bece78f4f4.png">

A little awkward, but it can work for now.